pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UnusedPrivateFieldRule.java

```diff
@@ -27,6 +32,17 @@
 
 public class UnusedPrivateFieldRule extends AbstractLombokAwareRule {
 
+    /**
+     * Fields, that are annotated with the annotation in the set
+     * should not be considered by the rule UnusedPrivateField.
+     */
+    private static final Set<Class<?>> NEGLECT_ANNOTATIONS_TYPE = new HashSet<>();
+
+    static {
+        NEGLECT_ANNOTATIONS_TYPE.add(java.lang.Deprecated.class);
+        NEGLECT_ANNOTATIONS_TYPE.add(javafx.fxml.FXML.class);
```

Comment:
this requires JavaFX to be in the classpath for pmd-java, which shouldn't... it's only used in pmd-ui (actually, that's the reason the build is currently failing).

Also of note, this reference can be different to the one used by the source code being analyzed (ie: different version), which would cause conflicts due to classloaders.

Consider just storing the string of the FQCN, and then using [`TypeHelper.isA()`](https://github.com/pmd/pmd/blob/master/pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/TypeHelper.java#L17)

Reply:
Thanks for your guidance!
Done.